internal/transport/http2_server.go

```diff
@@ -390,6 +390,13 @@ func (t *http2Server) operateHeaders(frame *http2.MetaHeadersFrame, handle func(
 			if timeout, err = decodeTimeout(hf.Value); err != nil {
 				headerError = true
 			}
+		// "Transports must consider requests containing the Connection header
+		// as malformed." - A41
+		case "connection":
```

Comment:
Should we be stripping this client-side as well?  @ejona86 

This is a behavior change and would impact users already using this header name, possibly accidentally.  This seems "okay", but just pointing it out.

Reply:
You cannot blindly strip, because you also need to remove any headers referenced from the `Connection` header. I think it is fine to just let the server break things. In Java we added a WARNING log when the Connection key is used because we were confident some of our internal clients were "doing it wrong" because we recently had trouble with Content-Length. https://github.com/grpc/grpc-java/pull/8559/files#diff-5788022e8c0c5de5ce8837a7ed2ff490f410cb31084c3811b6d1eeecb2e2ca1fR742-R745

You can see the necessary stripping code in cl/397767099.

I chose logging because I wanted to avoid doing the Connection header check every RPC and that code I put the logging in could be quite surprising if it started throwing.

Reply:
SGTM thanks.